src/main/java/pl/cleankod/ApplicationInitializer.java

```diff
@@ -34,18 +44,60 @@ AccountRepository accountRepository() {
     }
 
     @Bean
-    ExchangeRatesNbpClient exchangeRatesNbpClient(Environment environment) {
-        String nbpApiBaseUrl = environment.getRequiredProperty("provider.nbp-api.base-url");
-        return Feign.builder()
+    ExchangeRatesNbpClient exchangeRatesNbpClient(
+            @Value("${provider.nbp-api.base-url}") String nbpApiBaseUrl,
+            FeignDecorators feignDecorators
+    ) {
+        return Resilience4jFeign.builder(feignDecorators)
                 .client(new ApacheHttpClient())
                 .encoder(new JacksonEncoder())
                 .decoder(new JacksonDecoder())
+                .errorDecoder(new NbpClientErrorDecoder())
                 .target(ExchangeRatesNbpClient.class, nbpApiBaseUrl);
     }
 
     @Bean
-    CurrencyConversionService currencyConversionService(ExchangeRatesNbpClient exchangeRatesNbpClient) {
-        return new CurrencyConversionNbpService(exchangeRatesNbpClient);
+    FeignDecorators feignDecorators(
+            CircuitBreakerRegistry circuitBreakerRegistry
+    ) {
+        return FeignDecorators.builder()
+                .withCircuitBreaker(circuitBreakerRegistry.circuitBreaker("nbpClientCircuitBreaker"))
+//                .withFallbackFactory(ExchangeRatesNbpClientFallback::new)
+                .build();
+    }
+
+    @Bean
```

Comment:
do we really need to register it as a bean? Why not just a static factory method?